diff --git a/feeds.conf.default b/feeds.conf.default
index 8e14901..b597ac7 100644
--- a/feeds.conf.default
+++ b/feeds.conf.default
@@ -1,5 +1,5 @@
 src-git packages https://github.com/openwrt/packages.git;for-15.05
-src-git luci https://github.com/openwrt/luci.git;for-15.05
+src-git luci https://github.com/openwrt/luci.git
 src-git routing https://github.com/openwrt-routing/packages.git;for-15.05
 src-git telephony https://github.com/openwrt/telephony.git;for-15.05
 src-git management https://github.com/openwrt-management/packages.git;for-15.05
@@ -13,3 +13,11 @@ src-git management https://github.com/openwrt-management/packages.git;for-15.05
 #src-svn xfce svn://svn.openwrt.org/openwrt/feeds/xfce
 #src-svn lxde svn://svn.openwrt.org/openwrt/feeds/lxde
 #src-link custom /usr/src/openwrt/custom-feed
+src-git augsburg https://github.com/freifunk-augsburg/openwrt-packages.git
+src-git luci2 http://git.openwrt.org/project/luci2/ui.git
+src-git fastd git://git.metameute.de/lff/pkg_fastd
+src-git wbm git://github.com/battlemesh/battlemesh-packages.git
+src-git libreage git://github.com/libremap/libremap-agent-openwrt.git
+src-git kadnode git://github.com/mwarning/KadNode.git
+src-git kadlibsodium git://github.com/mwarning/libsodium-openwrt.git
+src-git fswebcam git://github.com/fsphil/fswebcam.git
